packages/terra-button/src/Button.jsx

```diff
@@ -115,225 +115,165 @@ const defaultProps = {
-    this.handleClick = this.handleClick.bind(this);
-
-    this.shouldShowFocus = true;
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (prevState.focused && nextProps.isDisabled) {
-      return { focused: false };
-    }
-    return null;
-  }
-
-  handleOnBlur(event) {
-    this.setState({ focused: false });
-
-    if (this.props.onBlur) {
-      this.props.onBlur(event);
+const Button = ({
+  icon,
+  isBlock,
+  isCompact,
+  isDisabled,
+  isIconOnly,
+  isReversed,
+  text,
+  type,
+  variant,
+  href,
+  onClick,
+  onMouseDown,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const handleOnBlur = (event) => {
+    if (onBlur) {
+      onBlur(event);
     }
-  }
+  };
 
-  handleClick(event) {
+  const handleClick = (event) => {
     // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
     // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
     // This will put focus on the button when clicked if it is not currently the active element.
     if (document.activeElement !== event.currentTarget) {
-      this.shouldShowFocus = false;
       event.currentTarget.focus();
-      this.shouldShowFocus = true;
     }
 
-    if (this.props.onClick) {
-      this.props.onClick(event);
+    if (onClick) {
+      onClick(event);
     }
-  }
+  };
 
-  handleKeyDown(event) {
+  const handleKeyDown = (event) => {
     // Add active state to FF browsers
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ active: true });
-
+    if (event.keyCode === KeyCode.KEY_SPACE) {
       // Follow href on space keydown when rendered as an anchor tag
-      if (this.props.href) {
-        // Prevent window scrolling
+      if (href) {
         event.preventDefault();
-        window.location.href = this.props.href;
+        window.location.href = href;
       }
     }
 
-    // Add focus styles for keyboard navigation
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
-      this.setState({ focused: true });
+    if (onKeyDown) {
+      onKeyDown(event);
     }
+  };
 
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
-    }
-  }
-
-  handleKeyUp(event) {
-    // Remove active state from FF broswers
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ active: false });
-    }
-
-    // Apply focus styles for keyboard navigation
-    if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
-      this.setState({ focused: true });
+  const handleKeyUp = (event) => {
+    if (onKeyUp) {
+      onKeyUp(event);
     }
+  };
 
-    if (this.props.onKeyUp) {
-      this.props.onKeyUp(event);
+  const handleFocus = (event) => {
+    if (onFocus) {
+      onFocus(event);
     }
-  }
+  };
```

Comment:
You don't need a separate method for this.  Since you are only checking for the existence of the onFocus callback, you can pass that function directly to the onFocus React event.  If it is undefined, the event just won't be created.  Same comment for the other handlers that do not have additional logic.